src/libraries/System.Private.CoreLib/src/System/IO/Path.cs

```diff
@@ -306,17 +285,9 @@ public static bool HasExtension(string? path)
 
         public static bool HasExtension(ReadOnlySpan<char> path)
         {
-            for (int i = path.Length - 1; i >= 0; i--)
-            {
-                char ch = path[i];
-                if (ch == '.')
-                {
-                    return i != path.Length - 1;
-                }
-                if (PathInternal.IsDirectorySeparator(ch))
-                    break;
-            }
-            return false;
+            int i = PathInternal.GetExtensionStart(path);
+            // Found and not last char
+            return (i > 0 && i < path.Length - 1);
```

Comment:
What about _hidden_ files like `.bashrc`? The dot is in position 0, but it's a file with an extension, just no name.

Reply:
Yeah this should be failing a test :)

Reply:
Is both filename and extension
![image](https://user-images.githubusercontent.com/1142958/97652489-65a8c100-1a56-11eb-9d6b-61c6c12c0a4f.png)

Reply:
Added test and fix